### DISCUSSION TITLE
ぜろちゃんねるプラス総合

### DISCUSSION THREAD

**Discussion by PrefKarafuto** [PrefKarafuto/ex0ch#10] [General]
ぜろちゃんねるプラスへの質問や、開発に関する意見・要望その他はこちらに。

**Comment by PYU224** [comment]
0ch+の管理画面のUIってどこのファイルが対応していますか？
手を入れようにもそこが分からないから手も足も出なくて申し訳ないです。

**Reply by PrefKarafuto** [reply]
ざっくり解説しますと、admin.cgiが掲示板設定情報のやユーザ情報の読み込みを行い、その後mordorフォルダのlogin.plでログイン処理を行ってsauron.plに飛ばします。これはいわゆる管理画面のメイン画面にあたり、ここから同フォルダ内のbbs.xxx.plやsys.xxx.pl、thread.xxx.plで、実際の各種操作(例えば掲示板作成・レスの削除等)を行います。この中でsub PrintXXXが、その機能の管理画面の表示を行い、sub FunctionXXXが管理機能処理を行います。sub DuPrintやDuFunctionはメニューの表示になります。

**Reply by PrefKarafuto** [reply]
![image](https://user-images.githubusercontent.com/122600023/212725080-29e9e9d5-5a5a-4f0f-b778-b5466c68f5e8.png)
これはサイドバーメニューの例ですが、ここにマウスオーバーしてもらうと、左下に
javascript:DoSubmit('xxx.xxx','DISP','XXX');
みたいな表示が出ると思うので、そこから各管理を処理するファイルがわかります。xxx.xxxがファイル名ですね。

**Reply by PYU224** [reply]
ありがとうございます。
鏡の国やYakumoBBSより難しいと感じていますが、どこに何のスクリプトがあるか分かれば何とかなりそうですね。
早速掲示板設定で不具合らしいものを見つけたので、プルリクエストしてよいですか？

**Reply by PrefKarafuto** [reply]
どしどしお願いします

**Reply by PYU224** [reply]
分かりました。早速やってみました。
これで大丈夫ですか？

**Comment by PrefKarafuto** [comment]
一応、呼ばれてるmoduleフォルダ内のモジュールの機能一覧です
before|after|機能  
---|---|---  
athelas     |plugin     |プラグイン管理  
baggins     |thread     |スレッド情報管理  
balrogs     |search     |検索  
celeborn    |archive    |過去ログ管理  
cidr_list   |cidr_list   |携帯IPのCIDRチェック  
constant    |constant    |定数の定義情報  
denethor    |banner    |バナー管理  
earendil    |file_utils    |ファイル操作ユーティリティ  
elves       |security       |管理セキュリティの管理  
faramir     |user     |アクセスユーザ管理  
galagriel   |data_utils   |汎用データ変換・取得  
gandalf     |notice     |ユーザ通知管理  
gondor      |dat      |datファイル管理  
httpservice |http_service |httpサービス  
imrahil     |log     |ログ管理  
isildur     |setting     |SETTINGデータ管理  
legolas     |header_footer_meta     |ヘッダ・フッタ・META管理  
melkor      |system      |システムデータ管理  
nazguls     |bbs_info     |掲示板情報管理  
newrelease  |update_notice  |アップデート通知  
orald       |error_info       |エラー情報管理モジュール  
peregrin    |manager_log    |管理ログデータの管理  
radagast    |cookie    |cookie管理  
samwise     |form     |フォーム情報管理  
session     |session     |セッション管理  
thorin      |buffer      |バッファ出力管理  
ungoliants  |cap  |キャップ管理  
vara        |post_service        |掲示板書き込み支援  
varda       |bbs_service       |bbs.cgi支援  
wormtongue  |ng_word  |NGワード管理

**Comment by PYU224** [comment]
鏡の国を改造してた際に思ったんですが、PHPに書き直す際に元のPerl版も残した方が良いかなと思っています。
理由はPHPはまだ開発途上の言語であり、バージョンアップ次第ではPHPにした掲示板スクリプトが使えなくなる可能性があるからです。
Perlならセットアップが難しい反面、枯れた技術だからスクリプトが付いていけなくなる事は少ないかと思います。

**Reply by PrefKarafuto** [reply]
理由もないですし、態々消さない限り基本的に残すとおもいます

**Reply by PYU224** [reply]
分かりました。

**Comment by PYU224** [comment]
今0ch+で検索機能を利用して気づいたんですが、大文字小文字の区別を付けて検索するんですね。
オプション等で大文字小文字の区別を付けず検索できても良いかもしれません。

**Comment by Kusaimara** [comment]
質問です。
不定期になりますがブログ等でこのプロジェクトの進捗報告ってしても大丈夫でしょうか？

**Reply by PrefKarafuto** [reply]
問題ありません

**Reply by Kusaimara** [reply]
分かりました。

**Reply by Kusaimara** [reply]
進捗を報告してみました。
https://kusaimara.com/2023/01/5chzerochplus.html

余談ですがアクセスカウンターを見ると先週に開発に参加した旨の下の記事
https://kusaimara.com/2023/01/zerochplus.html

のアクセス数が70あったので恒心教徒の人が半分だったとしても結構な数の一般の人が見てくださっているのだと思います。
なので機会を見つけては積極的に布教していこうと思います。

**Comment by PrefKarafuto** [comment]
0.7.5からのアップデート、chmateからの投稿が可能であることを確認しました

**Comment by PYU224** [comment]
ノヒケーやニヒケー、島田にかいの件で思ったんですが掲示板管理のログイン画面にもcaptcha認証を付けた方がいいかもしれません。

**Comment by PrefKarafuto** [comment]
てすや

**Reply by Ematojp** [reply]
てすや

**Comment by PrefKarafuto** [comment]
なんでも実況zero

**Comment by anime36** [comment]
Software error:
Undefined subroutine utf8::SWASHNEW called at ./admin/bbs.edit.pl line 586.
For help, please send mail to this site's webmaster, giving this error message and the time and date of the error.

Software error:
[Fri Oct 13 00:58:27 2023] admin.cgi: Undefined subroutine utf8::SWASHNEW called at ./admin/bbs.edit.pl line 586.
Compilation failed in require at /home/users/0/--/test/admin.cgi line 76.
For help, please send mail to this site's webmaster, giving this error message and the time and date of the error.

errorになるんですけど？

**Reply by PrefKarafuto** [reply]
修正しました
bbs.edit.plの586，587行目を
```
	$resmaxz 	=~ tr/0-9/０-９/; # 全角数字
	$resmaxz1 	=~ tr/0-9/０-９/; # 全角数字
```
と書き換えて下さい

**Reply by anime36** [reply]
同じエラーが出ました。

**Reply by PrefKarafuto** [reply]
こちらではエラー再現できないですね
0ch+バージョンは何を使っていますか？

**Reply by anime36** [reply]
New 0ch+ BBS :Build Version:0ch+ BBS 0.8.4 20230709

です

**Comment by anime36** [comment]
ワッチョイの導入が出来ませんでした。（プラグインは文字化けしてます）

**Reply by PrefKarafuto** [reply]
ワッチョイプラグインは本体に統合したため削除しました
Plugin(Perl_Unaltered)に置いてあるワッチョイプラグインは作業用のものですので、そのまま使うことはできません

**Reply by anime36** [reply]
BBS_SLIP（!slip:[vvv/vvvv/vvvvv/vvvvvv]）を有効にしたのですがコマンドが効きませんでした

**Reply by anime36** [reply]
この問題は自己解決しました。

専ブラでは問題無いのですがwebページから書き込みhttpsの書き込みページに飛ばない為cockの確認を一時停止しています

**Comment by PYU224** [comment]
防弾3Uでも報告しましたが、ここでも報告です。
https://w.atwiki.jp/vahren_ency/sp/pages/1.html
のようなアンダーバー付きのURLが、アンダーバーの途中から認識されない状態となっております。
すここで最初から報告してたら良かったですね。すみません。

**Comment by anime36** [comment]
Net::Whois::Rawが未定義になってるんですがどうすれば良いですか？

**Reply by PrefKarafuto** [reply]
エラー報告の際はバージョンとエラー画面を添付またはコピペしていただけると助かります

**Reply by anime36** [reply]
https://i.imgur.com/CPDVr1L.png

ex0ch Administrator Information
ex0ch BBS - Administrator Script

■Perl Packages (include perllib)
Net::Whois::Raw      | undefined
WP::Protocol::https | undefined

Build Version:0ch+ BBS dev-r124 20231030

**Reply by PrefKarafuto** [reply]
Net::Whois::Rawはbbsslip用に使われていましたが、処理が重くなったため暫定的に無効化しているものです
WP::Protocol::httpsはhCaptchaを使う場合に必要です
これらが必要な場合はcpanmなどでモジュールをインストールしてください

**Comment by PrefKarafuto** [comment]
色々あってなかなかv0.9.0出せてませんであれですが、忍法帖が完成次第出すことにします。

**Comment by anime36** [comment]
Software error:
Can't locate object method "fetch" via package "CGI::Cookie" (perhaps you forgot to load "CGI::Cookie"?) at ./module/ninpocho.pl line 84.
For help, please send mail to this site's webmaster, giving this error message and the time and date of the error.

https://i.imgur.com/TUILREG.png

ソフトウエアエラーがでました。

**Reply by PrefKarafuto** [reply]
nipocho.plの先頭部分のuseがある箇所に
```use CGI::Cookie;```
を加えてください

**Reply by anime36** [reply]
Can't locate object method "new" via package "Digest::MD5" (perhaps you forgot to load "Digest::MD5"?) at ./module/system.pl line 96.

**Reply by PrefKarafuto** [reply]
修正しました

**Comment by anime36** [comment]
perllib/Net/DNS.pm 行 41 の不明なメソッド「init」。
perllib/Net/DNS.pm 行 41 で require でコンパイルが失敗しました。
./module/data_utils.pl の 13 行目で、コンパイルが require で失敗しました。
BEGIN が失敗しました -- コンパイルは ./module/data_utils.pl 行 13 で中止されました。

Thu Dec 14 01:21:18 2023] admin.cgi: perllib/Net/DNS.pm 行 41 の不明なメソッド「init」。
[Thu Dec 14 01:21:18 2023] admin.cgi: perllib/Net/DNS.pm 行 41 で require でコンパイルが失敗しました。
[Thu Dec 14 01:21:18 2023] admin.cgi: ./module/data_utils.pl 行 13 の require でコンパイルが失敗しました。
[Thu Dec 14 01:21:18 2023] admin.cgi: BEGIN が失敗しました -- ./module/data_utils.pl の 13 行目でコンパイルが中止されました。
./admin/login.pl の 80 行目で、コンパイルが require で失敗しました。

ファイルを開けません: No such file or directory at ./module/slip.pl line 446.

**Reply by PrefKarafuto** [reply]
パスの設定がおかしかったので修正しました

**Reply by anime36** [reply]
Can't call method "is_empty" on an undefined value at ./module/ninpocho.pl line 123.

**Reply by PrefKarafuto** [reply]
まだ出ますか？

**Reply by anime36** [reply]
Software error:
Can't call method "is_empty" on an undefined value at ./module/ninpocho.pl line 123. 
![Screenshot_20231220-212233](https://github.com/PrefKarafuto/ex0ch/assets/147741952/31a742f7-dc1b-41ae-b157-fdd20fc264a9)

**Reply by PrefKarafuto** [reply]
CG::Sessionモジュールは入ってますか

**Reply by anime36** [reply]
忍法占を有効にするとerrorが出ます

**Reply by anime36** [reply]
![Screenshot_20231221-054521~2](https://github.com/PrefKarafuto/ex0ch/assets/147741952/be447314-1e85-47bf-bdd2-008dfe37a324)

**Comment by anime36** [comment]
Software error:

ファイルを開けません: No such file or directory at ./module/slip.pl line 446. 

ソフトウェアエラーが出ます

**Reply by anime36** [reply]
![Screenshot_20240116-165227](https://github.com/PrefKarafuto/ex0ch/assets/147741952/b60b647a-e7f9-4def-aafd-c8a7440bfc79)

ディレクトリ一覧です

**Reply by anime36** [reply]
旧versionに戻したら書き込めるようになりました

**Comment by anime36** [comment]
Can't use an undefined value as a SCALAR reference at ./module/admin_search.pl line 302, <$fh> line 440.

なんかエラーでました。

**Reply by PrefKarafuto** [reply]
まだ出ますか？

**Reply by anime36** [reply]
もうでないです

**Comment by anime36** [comment]
./module/post_service.pl 行 519 で未定義の値に対してメソッド "Get" を呼び出すことができません。
